small_text/integrations/transformers/classifiers/classification.py

```diff
@@ -122,32 +122,55 @@ def __init__(self,
 
 
 def _get_layer_params(model, base_lr, fine_tuning_arguments):
-
     layerwise_gradient_decay = fine_tuning_arguments.layerwise_gradient_decay
 
     params = []
 
+    # Get layers under assumption that a certain naming convention is kept
     base_model = getattr(model, model.base_model_prefix)
+    layers = []
+    if hasattr(base_model, 'embeddings'):
```

Comment:
Yeah I also thought of using named_children but the problem is here that, to the best of my knowledge, they are given in the ordering as they are defined in the model not in the one how they are connected, right? Which, in my mind at least, can be even more arbitrary than the naming convention (because no one will blame you if you defined the encoder after the classifier as long as you connected them the right way, right? But if one named them weirdly they might get frowned upon) therefore one might need to redefine the loop in something that uses the names again which would bring us back to the start?

Plus: Changing the names is often done with great care because that makes versions easily incompatible with each other, reordering on the other hand can be done super easily and can therefore change quite swiftly

Reply:
Ugh, good point. This leaves us with very few options for a generic mechanism:
1. We assume a fixed setup where certain modules have to be present and whose names have to be what we expect them to be.
2. We provide this per model (which would be a lot of effort).

In this case, I have to go back and support your solution (1.) to check for fixed names.